Sources/SwiftExtensions/Atomics.swift

```diff
@@ -10,97 +10,80 @@
 //
 //===----------------------------------------------------------------------===//
 
-import CAtomics
+import Synchronization
 
-// TODO: Use atomic types from the standard library (https://github.com/swiftlang/sourcekit-lsp/issues/1949)
 package final class AtomicBool: Sendable {
```

Comment:
I think we should remove the `AtomicBool` etc. types altogether and just use `Atomic<Bool>` instead. Their purpose was to wrap the C functions, which isn’t needed anymore.

Reply:
I actually tried that but I failed to implement it since Atomic is non-copyable and thus we can't pass it into tasks for example:

``` swift
 let didHitTimeout = Atomic<Bool>(false)

  let stream = AsyncThrowingStream<T?, Error> { continuation in
  // Passing closure as a 'sending' parameter risks causing data races between code in the current task and concurrent execution of the closure
    Task {
      didHitTimeout.store(true, ordering: ...)
    }
  }
```

Reply:
Could we just store it in a class to get reference semantics in those cases?

```class
final class Wrapped<T: ~Copyable & Sendable>: Sendable {
  let value: T

  init(_ value: consuming T) {
    self.value = value
  }
}
```

in those cases? I would expect that in most cases we can still use `Atomic<Bool>` directly.

Reply:
Should be better now